Dalamud/Interface/ImGuiNotification/Internal/ActiveNotification.cs

```diff
@@ -0,0 +1,355 @@
+    }
+
+    /// <inheritdoc/>
+    public float Progress
+    {
+        get => this.newProgress ?? this.underlyingNotification.Progress;
+        set => this.newProgress = value;
+    }
+
+    /// <summary>Gets the eased progress.</summary>
+    private float ProgressEased
+    {
+        get
+        {
+            var underlyingProgress = this.underlyingNotification.Progress;
+            if (Math.Abs(underlyingProgress - this.progressBefore) < 0.000001f || this.progressEasing.IsDone)
+                return underlyingProgress;
+
+            var state = Math.Clamp((float)this.progressEasing.Value, 0f, 1f);
+            return this.progressBefore + (state * (underlyingProgress - this.progressBefore));
+        }
+    }
+
+    /// <summary>Gets the string for the initiator field.</summary>
+    private string InitiatorString =>
+        this.initiatorPlugin is not { } plugin
+            ? NotificationConstants.DefaultInitiator
+            : this.isInitiatorUnloaded
+                ? NotificationConstants.UnloadedInitiatorNameFormat.Format(plugin.Name)
+                : plugin.Name;
+
+    /// <summary>Gets the effective text to display when minimized.</summary>
+    private string EffectiveMinimizedText => (this.MinimizedText ?? this.Content).ReplaceLineEndings(" ");
+
+    /// <inheritdoc/>
+    public void DismissNow() => this.DismissNow(NotificationDismissReason.Programmatical);
+
+    /// <summary>Dismisses this notification. Multiple calls will be ignored.</summary>
+    /// <param name="reason">The reason of dismissal.</param>
+    public void DismissNow(NotificationDismissReason reason)
+    {
+        if (this.DismissReason is not null)
+            return;
+
+        this.DismissReason = reason;
+        this.hideEasing.Start();
+        this.InvokeDismiss();
+    }
+
+    /// <inheritdoc/>
+    public void ExtendBy(TimeSpan extension)
+    {
+        var newExpiry = DateTime.Now + extension;
+        if (this.extendedExpiry < newExpiry)
+            this.extendedExpiry = newExpiry;
+    }
+
+    /// <inheritdoc/>
+    public void SetIconTexture(IDalamudTextureWrap? textureWrap)
+    {
+        this.SetIconTexture(textureWrap is null ? null : Task.FromResult(textureWrap));
+    }
+
+    /// <inheritdoc/>
+    public void SetIconTexture(Task<IDalamudTextureWrap?>? textureWrapTask)
+    {
+        if (this.DismissReason is not null)
+        {
+            textureWrapTask?.ToContentDisposedTask(true);
+            return;
+        }
+
+        // After replacing, if the old texture is not the old texture, then dispose the old texture.
+        if (Interlocked.Exchange(ref this.iconTextureWrap, textureWrapTask) is { } wrapTaskToDispose &&
+            wrapTaskToDispose != textureWrapTask)
+        {
+            wrapTaskToDispose.ToContentDisposedTask(true);
+        }
+    }
+
+    /// <summary>Removes non-Dalamud invocation targets from events.</summary>
+    internal void RemoveNonDalamudInvocations()
```

Comment:
Maybe it would be good to leave a note here as to why this is required(=> we don't want to remove notifications when a plugin goes away, so we just make the actions do nothing?)